iam-login-service/src/main/java/it/infn/mw/iam/core/expression/IamSecurityExpressionMethods.java

```diff
@@ -153,4 +161,10 @@ public boolean hasDashboardRole(Role role) {
   public boolean hasAdminOrGMDashboardRoleOfGroup(String gid) {
     return (hasDashboardRole(Role.ROLE_ADMIN) || isGroupManager(gid));
   }
+
+  public boolean isClientOwner(String clientId) {
+    Optional<IamAccount> owner = accountUtils.getAuthenticatedUserAccount();
+    Optional<ClientDetailsEntity> client = clientRepo.findByClientId(clientId);
+    return owner.isPresent() && client.isPresent() && accountClientRepo.findByAccountAndClient(owner.get(), client.get()).isPresent();
```

Comment:
Please split this triple condition in different statements to make it more readable. See https://github.com/zakirullin/cognitive-load?tab=readme-ov-file#complex-conditionals